### ISSUE TITLE
terragrunt-info as internal function

### ISSUE THREAD

**Issue by antonbabenko** [gruntwork-io/terragrunt#750]
It would be great to be able to access values of `terragrunt terragrunt-info` using a function or via env vars:
```terraform
after_hook "copy_common_main_providers" {
  commands = ["init-from-module"]
  execute  = ["cp", "something.tf", get_terragrunt_info("WorkingDir")]
}
```

**Comment by ekini** [comment]
Probably can get away with something like `get_terragrunt_workingdir()`?
I addressed it https://github.com/gruntwork-io/terragrunt/issues/736#issuecomment-500672816, and maybe https://github.com/gruntwork-io/terragrunt/issues/736#issuecomment-501007842 can be a possible (and better) solution?

**Comment by antonbabenko** [comment]
Yes, it would be very helpful to have a function like `get_terragrunt_workingdir()` which returns `/var/www/eu-west-1/vpc/.terragrunt-cache/8tw6nJm13wWcoe79ee6cpyG1tvk/BoKn_3NNiOJEg4PmxaP5wYFi3M8`.

@brikis98 what do you think?

**Comment by ekini** [comment]
@antonbabenko As a workaround maybe you can symlink `something.tf` in the module?

**Comment by antonbabenko** [comment]
Meanwhile, I run another hook at the end to remove the file in the wrong location.

**Comment by brikis98** [comment]
Adding new variables and functions to Terragrunt is easy now: you just add it to the [EvalContext](https://github.com/gruntwork-io/terragrunt/blob/master/config/config_helpers.go#L94-L96). The earlier versions of Terragunt solely added functions, as those were easier to find with our ugly regex, but now that all of this is handled in a first-class way by the HCL2 parser, we can use variables or functions, depending on what makes sense. PRs welcome.

**Comment by rgarrigue** [comment]
@antonbabenko do you mind sharing your cleaning hook ? Can't seems to get mine right, it's also nuking files in the cache dir, hence I can't run terragrunt twice in a row.
